### DISCUSSION TITLE
Team Radio ideas

### DISCUSSION THREAD

**Discussion by ely105** [f1multiviewer/issue-tracker#356] [Feature requests]
I tried to look through all of the ideas to see if this one is out there.  Maybe I missed it.  There may be some interdependencies so it may not fit neatly into one idea, but maybe a good, better, best progression.  I can break them out if thats better.

VU Meters: (small/corner of window)
- For those that would like to monitor more than 1 team radio, it'd be nice to have some visual representation of when a radio has active conversation on it.  That way you can more easily identify which radio you might be hearing.  Now I realize that there is background engine noise, so there may need to be some adjustment possible to control how sensitive the VU meters are.  It seems like the vocals are usually louder than the engine noise, so maybe raising/setting the floor level so that the meters show more movement when there is chatter.

multi-Radio priority/muting:
- If you are listening to a few radios there are occasions where everyone is talking at once, so when 2 or 3 are talking it pretty much makes them all unintelligable.  Maybe a user-set priority to when this occurs that mutes/pads lower priority radios.

Background noise removal:
- I'm wondering if something like Krisp.ai (or something else) could be used to elminate much of the background engine noise so that we hear more clean speech. (at least on the driver side) It sounds to me like some drivers use PTT while others leave open audio and you hear the constant engine noise.  I know Krisp.ai have an audio SDK, but not sure how taxing it is on a per stream basis.

Radio live captioning:
- If you can clean up the audio with background noise removal, then it opens the door for maybe realtime captioning of the audio stream.  That could be useful to monitor chatter without having to listen to many streams.

I've heard others suggest radio panning which i think is a good idea also.  prolly good for up to 3 radios if you do left/center/right.  you could get into 3d surround but that seems like too much.  I think 2-3 is probably the max you could expect to listen to, but captions could really open up the ability to monitor more.

**Comment by jeppeolesen** [comment]
Hey @ely105 

The live captioning has been implemented in the app and can be accessed via the live timing panel. We're not doing any processing of the feeds coming from F1 TV, so using something like Krispr is out of scope. I know that F1 TV themselves are working on similar solutions, so we'll leave it up to them to clean up the audio 😅 

As for the VU meter and radio priority, could you submit them as separate feature requests? That makes it a lot easier for us to process them.
